pandas/tests/io/excel/test_style.py

```diff
@@ -123,6 +145,39 @@ def test_styler_to_excel_unstyled(engine):
 ]
 
 
+@pytest.mark.parametrize(
+    "css",
+    ["background-color: #111222"],
+)
+def test_styler_custom_style(css):
+    # GH 54154
+    openpyxl = pytest.importorskip("openpyxl")
+    df = DataFrame([{"A": 1, "B": 2}, {"A": 1, "B": 2}])
+
+    with tm.ensure_clean(".xlsx") as path:
+        with ExcelWriter(path, engine="openpyxl") as writer:
+            styler = df.style.map(lambda x: css)
+            styler.to_excel(writer, sheet_name="custom", index=False)
+
+        with contextlib.closing(openpyxl.load_workbook(path)) as wb:
+            # Check font, spacing, indentation
+            assert wb["custom"].cell(1, 1).font.bold is False
+            assert wb["custom"].cell(1, 1).alignment.horizontal is None
+            assert wb["custom"].cell(1, 1).alignment.vertical is None
+
+            # Check border
+            assert wb["custom"].cell(1, 1).border.bottom.color is None
+            assert wb["custom"].cell(1, 1).border.top.color is None
+            assert wb["custom"].cell(1, 1).border.left.color is None
+            assert wb["custom"].cell(1, 1).border.right.color is None
+
+            # Check background color
+            assert wb["custom"].cell(2, 1).fill.fgColor.index == "00111222"
+            assert wb["custom"].cell(3, 1).fill.fgColor.index == "00111222"
+            assert wb["custom"].cell(2, 2).fill.fgColor.index == "00111222"
+            assert wb["custom"].cell(3, 2).fill.fgColor.index == "00111222"
+
+
```

Comment:
I think we can simplify code test by combining both functions test_styler_default_values and test_styler_custom_style to one function.
```
@pytest.mark.parametrize(
    "css, color",
    [("", "00000000"), 
     ("background-color: #111222", "00111222")],
)
def test_default_and_custom_style(css, color):
    # GH 54154
    openpyxl = pytest.importorskip("openpyxl")
    df = DataFrame([{"A": 1, "B": 2}, {"A": 1, "B": 2}])

    with tm.ensure_clean(".xlsx") as path:
        with ExcelWriter(path, engine="openpyxl") as writer:
            styler = df.style.map(lambda x: css)
            styler.to_excel(writer, sheet_name="custom", index=False)

        with contextlib.closing(openpyxl.load_workbook(path)) as wb:
            # Check font, spacing, indentation
            assert wb["custom"].cell(1, 1).font.bold is False
            assert wb["custom"].cell(1, 1).alignment.horizontal is None
            assert wb["custom"].cell(1, 1).alignment.vertical is None

            # Check border
            assert wb["custom"].cell(1, 1).border.bottom.color is None
            assert wb["custom"].cell(1, 1).border.top.color is None
            assert wb["custom"].cell(1, 1).border.left.color is None
            assert wb["custom"].cell(1, 1).border.right.color is None

            # Check background color
            assert wb["custom"].cell(2, 1).fill.fgColor.index == color
            assert wb["custom"].cell(3, 1).fill.fgColor.index == color
            assert wb["custom"].cell(2, 2).fill.fgColor.index == color
            assert wb["custom"].cell(3, 2).fill.fgColor.index == color
```